compiler/rustc_expand/src/build.rs

```diff
@@ -467,6 +499,17 @@ impl<'a> ExtCtxt<'a> {
         self.expr(span, ast::ExprKind::If(cond, self.block_expr(then), els))
     }
 
+    // if !{ ... } { ... } else { ... }
+    pub fn expr_if_not(
+        &self,
+        span: Span,
+        cond: P<ast::Expr>,
+        then: P<ast::Expr>,
+        els: Option<P<ast::Expr>>,
+    ) -> P<ast::Expr> {
+        self.expr_if(span, self.expr(span, ast::ExprKind::Unary(UnOp::Not, cond)), then, els)
+    }
```

Comment:
This whole AST builder interface should be moved to `rustc_builtin_macros`, so I recommend adding the new required functions directly to `compiler/rustc_builtin_macros/src/assert/context.rs` instead of extending it.